sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -469,12 +469,9 @@ export class StudentService extends RecordDataModelService<Student> {
           "application",
           "application.student.id = student.id",
         )
-        .leftJoin("application.location", "pirLocation")
-        .leftJoin("pirLocation.institution", "pirInstitution")
-        .leftJoin("application.currentAssessment", "studentAssessment")
-        .leftJoin("studentAssessment.offering", "offering")
-        .leftJoin("offering.institutionLocation", "offeringLocation")
-        .leftJoin("offeringLocation.institution", "offeringInstitution");
+        .leftJoin("application.location", "institutionLocation")
+        .leftJoin("institutionLocation.institution", "institution")
+        .leftJoin("application.currentAssessment", "studentAssessment");
```

Comment:
@andrepestana-aot @andrewsignori-aot @ann-aot @guru-aot 

Considering the context of the story, is it not sufficient to consider  a student having an application submitted with it's location->institution matching the institution who is performing the search and application not being in overwritten status? if my assumption is right we can ignore the join with currentAssessment right? let me know if I am missing something. 

![image](https://user-images.githubusercontent.com/54600590/232162256-b04ad0f7-1ffc-4a12-b751-357b7eb675e9.png)

Reply:
@dheepak-aot if your suggestion is about checking the `application.location` and, because `application.location` is only set for submitted applications, we can rely on that, it may be acceptable.
Checking if the assessment record was present is juts, for instance, to exclude a possible canceled draft application.

Reply:
The join with currentAssessment if only for the case of the cancelled applications. The student can create a draft application and cancel it and we want to avoid this case to be found by the institution. So, to find only the cancelled ones that have been submitted, we check if the application is cancelled AND had an assessment.

Reply:
@andrepestana-aot if the application is created and canceled while in the draft we are not setting the `application.location`, right?

Reply:
@andrewsignori-aot @andrepestana-aot A cancelled draft application will not have locationId right? as it is populated on submission, so it will be automatically excluded right?

Let me know if I am missing something.

Reply:
@dheepak-aot this was exactly what I meant to say with my comment above, so yes, I believe that we are on the same page 😉

Reply:
@dheepak-aot I agree. I was also in the same page 👍

Reply:
I was adding the locationId as a Draft too, but I see what you guys are saying and I'm going to change then and change the query as well. Thanks!

Reply:
@andrepestana-aot were you adding it to the draft? I saw the code doing it on the transition from "Draft->Submit" and this will not change. The application draft is saved in a different API endpoint.

Reply:
yes, you're right @andrewsignori-aot . I got confused.